home-assistant-plugin/custom_components/xknx/sensor.py

```diff
@@ -13,21 +14,27 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
     entities = []
     for device in hass.data[DOMAIN].xknx.devices:
         if isinstance(device, XknxSensor):
-            entities.append(KNXSensor(device))
+            entities.append(KNXSensor(device, hass))
     async_add_entities(entities)
 
 
 class KNXSensor(KnxEntity, Entity):
     """Representation of a KNX sensor."""
 
-    def __init__(self, device: XknxSensor):
+    def __init__(self, device: XknxSensor, hass: HomeAssistant):
         """Initialize of a KNX sensor."""
+        if device.value_template is not None:
+            device.value_template.hass = hass
         super().__init__(device)
 
     @property
     def state(self):
         """Return the state of the sensor."""
-        return self._device.resolve_state()
+        state = self._device.resolve_state()
+        template = self._device.value_template
+        if template is not None and state is not None:
+            state = template.async_render_with_possible_json_value(state)
```

Comment:
I'd say json isn't possible here. You could call async_render() directly.
https://github.com/home-assistant/core/blob/960b5b7d86dcb32a720b8e87e5403df9e5ae466b/homeassistant/helpers/template.py#L342

Reply:
I have tried to get `async_render()` running, without success. Unfortunately I did not found any example how to implement `async_render()`.  I have found some examples [here ](https://github.com/home-assistant/core/blob/ee914366a8762f495a585e09a0bd7b0675b4011c/tests/helpers/test_script_variables.py) and in other components, but I fear that I have too less background knowledge.

Reply:
Solved by 4be04b4